rest/tests/test_rest.py

```diff
@@ -0,0 +1,742 @@
+
+        terms = self.get_terms(self.repo.slug, vocab.slug)
+        self.assertEqual(1, terms['count'])
+        terms = self.get_terms(self.repo.slug, vocab2.slug)
+        self.assertEqual(1, terms['count'])
+
+        self.get_term(self.repo.slug, vocab.slug, term1['slug'])
+        self.get_term(self.repo.slug, vocab.slug, term2['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term1['slug'],
+                      expected_status=NOT_FOUND)
+        self.get_term(self.repo.slug, vocab2.slug, term2['slug'])
+
+    def test_vocabulary_pagination(self):
+        """Test pagination for collections"""
+
+        expected = [
+            create_vocabulary(
+                self.repo.id,
+                "name{i}".format(i=i),
+                "description",
+                True,
+                Vocabulary.FREE_TAGGING,
+                1000
+            ) for i in range(100)]
+
+        resp = self.client.get(
+            '/api/v1/repositories/{repo_slug}/vocabularies/'.format(
+                repo_slug=self.repo.slug,
+            ))
+        self.assertEqual(OK, resp.status_code)
+        vocabularies = as_json(resp)
+        self.assertEqual(100, vocabularies['count'])
+        self.assertEqual([VocabularySerializer(x).data for x in expected[:20]],
+                         vocabularies['results'])
+
+        resp = self.client.get(
+            '/api/v1/repositories/{repo_slug}/vocabularies/?page=2'.format(
+                repo_slug=self.repo.slug,
+            ))
+        self.assertEqual(OK, resp.status_code)
+        vocabularies = as_json(resp)
+        self.assertEqual(100, vocabularies['count'])
+        self.assertEqual([VocabularySerializer(x).data
+                          for x in expected[20:40]], vocabularies['results'])
+
+    def test_authentication(self):
+        """Make sure unauthenticated and unauthorized users can't get access"""
+
+        anonymous = Client()
+
+        resp = anonymous.get(
+            '/api/v1/repositories/{slug}/'.format(
+                slug=self.repo.slug
+            )
+        )
+        self.assertEqual(FORBIDDEN, resp.status_code)
+
+        client = Client()
+        client.login(
+            username=self.USERNAME_NO_REPO,
+            password=self.PASSWORD
+        )
+
+        resp = self.client.get(
+            '/api/v1/repositories/{slug}/'.format(
+                slug=self.repo.slug
+            )
+        )
+        self.assertEqual(OK, resp.status_code)
+        repo_dict = as_json(resp)
+        self.assertEqual(self.repo.name, repo_dict['name'])
+
+        repo_dict['name'] = 'another name'
+        resp = client.post(
+            '/api/v1/repositories/', repo_dict
+        )
+        self.assertEqual(FORBIDDEN, resp.status_code)
+
+        resp = anonymous.post(
+            '/api/v1/repositories/', repo_dict
+        )
+        self.assertEqual(FORBIDDEN, resp.status_code)
+
```

Comment:
These functions all use self.client but I want to test it using an anonymous user